<a href="https://colab.research.google.com/github/ZubairQazzi/Catch-Color-Ball-Game-/blob/main/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/ZubairQazzi/flyrank-ml-internship-zubair/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

My provisional lane is Refresh / Content Opportunity Scoring. I chose this lane because the starter notebooks showed how search and content signals can be used to rank pages for review. Over the next seven weeks, I want to explore which pages should be reviewed first for refresh, protection, expansion, pruning, or monitoring. My goal is to create a transparent ranked list using measurable signals while keeping the final decision with a human reviewer.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Research question: Which content pages should be reviewed first for refresh, protection, expansion, pruning, or monitoring based on their search visibility, traffic, age, engagement, and trend signals?

Unit of analysis: One content page.

Output: A ranked list of pages with a suggested action, confidence level, and short reason for each recommendation.

Decision: Which pages should receive attention first when the team has limited time.

Action: A human reviewer checks the highest-ranked pages and decides whether to refresh, expand, protect, prune, or monitor them.

Cost of a wrong recommendation: A false positive could waste time on a page that does not need work, while a false negative could cause the team to miss an important declining or high-opportunity page.

Why data or ML can help: Reviewing every page manually is slow. Data and simple machine-learning methods can combine several signals and help prioritize pages, but the final decision should remain with a human reviewer.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import pandas as pd

data_url = (
    "https://raw.githubusercontent.com/"
    "ZubairQazzi/flyrank-ml-internship-zubair/"
    "main/data/raw/content_refresh_anonymized.csv"
)

df = pd.read_csv(data_url)

total_pages = len(df)

declining_with_demand = (
    (df["trend_direction"] == "down")
    & (df["impressions_90d"] >= 100)
).sum()

high_visibility_pages = (
    df["impressions_90d"] >= 500
).sum()

print(f"Total pages in the starter dataset: {total_pages:,}")
print(f"Declining pages with at least 100 impressions: {declining_with_demand:,}")
print(f"Pages with at least 500 impressions: {high_visibility_pages:,}")

Total pages in the starter dataset: 30,000
Declining pages with at least 100 impressions: 13,152
Pages with at least 500 impressions: 16,726


The starter dataset contains 30,000 pages. Among them, 13,152 pages are declining while still receiving at least 100 impressions, and 16,726 pages have at least 500 impressions. These numbers suggest there are many visible pages that may need prioritised review, making Refresh / Content Opportunity Scoring a useful lane to explore.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project will provide an observed and directional ranking of pages that may deserve review based on measurable search, traffic, engagement, age, and trend signals. The results will support a human reviewer in deciding which pages to examine first. They will not prove that refreshing a page caused better performance, guarantee that a recommended action will work, or reveal how Google’s algorithm operates.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.